drivers/mock/driver.go

```diff
@@ -358,21 +367,39 @@ func (d *Driver) RecoverTask(handle *drivers.TaskHandle) error {
 		return fmt.Errorf("failed to decode task state from handle: %v", err)
 	}
 
-	driverCfg, err := parseDriverConfig(handle.Config)
-	if err != nil {
-		d.logger.Error("failed to parse driver config from handle", "error", err, "task_id", handle.Config.ID, "config", hclog.Fmt("%+v", handle.Config))
-		return fmt.Errorf("failed to parse driver config from handle: %v", err)
+	taskState.Command.parseDurations()
+	if taskState.ExecCommand != nil {
+		taskState.ExecCommand.parseDurations()
 	}
 
-	// Remove the plugin exit time if set
-	driverCfg.pluginExitAfterDuration = 0
-
 	// Correct the run_for time based on how long it has already been running
 	now := time.Now()
-	driverCfg.runForDuration = driverCfg.runForDuration - now.Sub(taskState.StartedAt)
+	if !taskState.StartedAt.IsZero() {
+		taskState.Command.runForDuration = taskState.Command.runForDuration - now.Sub(taskState.StartedAt)
+
+		if taskState.ExecCommand != nil {
+			taskState.ExecCommand.runForDuration = taskState.ExecCommand.runForDuration - now.Sub(taskState.StartedAt)
+		}
+	}
+
+	// Recreate the taskHandle. Because there's no real running process, we'll
+	// assume we're still running if we've recovered it at all.
+	killCtx, killCancel := context.WithCancel(context.Background())
+	h := &taskHandle{
+		logger:          d.logger.With("task_name", handle.Config.Name),
+		pluginExitAfter: taskState.PluginExitAfter,
+		killAfter:       taskState.KillAfter,
+		waitCh:          make(chan any),
+		taskConfig:      handle.Config,
+		command:         taskState.Command,
+		execCommand:     taskState.ExecCommand,
+		//		procState:       drivers.TaskStateRunning,
```

Comment:
we don't want this?

Reply:
Oops, that's left over from debugging the test to make sure we don't _need_ to set it, because it should get set when we call `go h.run()` below and that'll account for when the `run_for` expires. However, we also should be setting it here for fidelity with the other task drivers. Will fix.

Reply:
Done. In order to make sure the test doesn't just read that value, I also had to change it from `InitialSuccess` to `ContinualSuccess`